src/clickonce/launcher/Constants.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Deployment.Launcher
+{
+    public static class Constants
+    {
+        // Logging - registry keys/values
+        public static readonly string ClickOnceDeploymentSubkeyName = @"SOFTWARE\Classes\Software\Microsoft\Windows\CurrentVersion\Deployment";
+        public static readonly string LauncherLogFilePathRegistryString = "LauncherLogFilePath";
+
+        // Error messages
+        public static readonly string ErrorApplicationFilenameCannotBeEmpty = "Filenname cannot be empty.";
```

Comment:
These really should be in a localized resource. This is a shipping gate typically.

Reply:
Launcher has no UI and strings are intended for debugging purpose only. I will inquire about the loc requirement.

Reply:
If these strings are never shown to users I think this might be okay. Thanks for checking though.

Reply:
The strings will be logged in the optional log-file specified by user - does this still fall under "never shown to users"?

Reply:
That is getting into one of those fuzzy areas. Let me help you find an authority offline.

Reply:
If the strings are internal only and not for users then localization is not essential IMO. By the way "filename" has a spelling mistake :)

Reply:
The strings will only be logged if user sets logging in registry. It could help investigate failures to launch apps.

Yeah - I noticed the typo :) It was fixed in second commit.

Reply:
I will create a new issue to discuss localization requirement.